src/sentry/utils/auth.py

```diff
@@ -338,43 +348,45 @@ def log_auth_success(request, username, organization_id=None, source=None):
     )
 
 
-def log_auth_failure(request, username=None):
+def log_auth_failure(request: HttpRequest, username: str | None = None) -> None:
     logger.info(
         "user.auth.fail", extra={"ip_address": request.META["REMOTE_ADDR"], "username": username}
     )
 
 
-def has_user_registration():
+def has_user_registration() -> bool:
     from sentry import features, options
 
     return features.has("auth:register") and options.get("auth.allow-registration")
 
 
-def is_user_signed_request(request):
+def is_user_signed_request(request: HttpRequest) -> bool:
     """
     This function returns True if the request is a signed valid link
     """
     try:
-        return request.user_from_signed_request
+        return bool(request.user_from_signed_request)
     except AttributeError:
         return False
 
 
-def set_active_org(request: Request, org_slug: str) -> None:
+def set_active_org(request: HttpRequest, org_slug: str) -> None:
     # even if the value being set is the same this will trigger a session
     # modification and reset the users expiry, so check if they are different first.
     if hasattr(request, "session") and request.session.get("activeorg") != org_slug:
         request.session["activeorg"] = org_slug
 
 
-class EmailAuthBackend(ModelBackend):
+class EmailAuthBackend(ModelBackend):  # type: ignore
```

Comment:
Is the type ignore because the super class is from a file not in mypy?  Super unfortunate.

Reply:
Yes. Not in Mypy, and moreover a Django class. It would be nice if those were typesafe, which apparently is [Django issue #29299](https://code.djangoproject.com/ticket/29299).